reader.go

```diff
@@ -348,7 +352,7 @@ func decodeLineOfMasterPlaylist(p *MasterPlaylist, state *decodingState, line st
 }
 
 // Parse one line of media playlist.
-func decodeLineOfMediaPlaylist(p *MediaPlaylist, wv *WV, state *decodingState, line string, strict bool) error {
+func decodeLineOfMediaPlaylist(p *MediaPlaylist, wv *WV, state *decodingState, line string, strict, sharedState bool) error {
```

Comment:
I see what's happening here, but I found the previous iteration to be much more readable.

This may all be correct, but I am not easily able to understand this pull request without looking back at the previous iteration.

Would you be open to reverting back to it?

Reply:
> Would you be open to reverting back to it?

Sorry, didn't get - revert to what point?

> I see what's happening here, but I found the previous iteration to be much more readable.

Here I've just fixed tests (not mine) which failed because of my addition of new error.
(It was `TestDecodeMediaPlaylistWithAutodetection` that fails). It was failing because `m3u8.Decode` function shares state between `decodeLineOfMasterPlaylist` and `decodeLineOfMediaPlaylist`, so I've just made special path for that case.

Reply:
Please force-push back to a63f94a and then apply your incremental changes on top of that.   The force-push broke my continuity of review.  The current state of code is less readable than what I reviewed earlier

Reply:
done

Reply:
thank you for that 👍